cmd/crossplane/core/core.go

```diff
@@ -59,11 +60,13 @@ func (c *Command) Run() error {
 }
 
 type startCommand struct {
-	Namespace      string `short:"n" help:"Namespace used to unpack and run packages." default:"crossplane-system" env:"POD_NAMESPACE"`
-	CacheDir       string `short:"c" help:"Directory used for caching package images." default:"/cache" env:"CACHE_DIR"`
-	LeaderElection bool   `short:"l" help:"Use leader election for the controller manager." default:"false" env:"LEADER_ELECTION"`
-	Registry       string `short:"r" help:"Default registry used to fetch packages when not specified in tag." default:"${default_registry}" env:"REGISTRY"`
-	CABundlePath   string `help:"Additional CA bundle to use when fetching packages from registry." env:"CA_BUNDLE_PATH"`
+	Namespace            string `short:"n" help:"Namespace used to unpack and run packages." default:"crossplane-system" env:"POD_NAMESPACE"`
+	CacheDir             string `short:"c" help:"Directory used for caching package images." default:"/cache" env:"CACHE_DIR"`
+	LeaderElection       bool   `short:"l" help:"Use leader election for the controller manager." default:"false" env:"LEADER_ELECTION"`
+	Registry             string `short:"r" help:"Default registry used to fetch packages when not specified in tag." default:"${default_registry}" env:"REGISTRY"`
+	CABundlePath         string `help:"Additional CA bundle to use when fetching packages from registry." env:"CA_BUNDLE_PATH"`
+	WebhookTLSSecretName string `help:"The name of the TLS Secret that will be used by the webhook servers of core Crossplane and providers." env:"WEBHOOK_TLS_SECRET_NAME"`
+	WebhookTLSCertDir    string `help:"The directory of TLS certificate that will be used by the webhook server of core Crossplane. There should be tls.crt and tls.key files." env:"WEBHOOK_TLS_CERT_DIR"`
```

Comment:
Will these be considered an alpha feature initially? If so, should this be included in the `Alpha Features` group?

Reply:
I thought having them optional would be enough for what we do for alpha features, i.e. disabled unless explicitly enabled. In general, not sure if I'd consider them alpha because we're only enabling what's possible in controller-runtime and k8s, rather than introduce a new field or type to users for example. And there are cases where you have to enable them, like providers with CRDs whose conversion strategy is webhook.

Reply:
It's probably best to be conservative here - I'd probably err toward an `--enable-webhooks` flag in the "Alpha Features" group even if it was just for consistency.

Perhaps I could be convinced it's not needed though. Will this PR result in any practical changes to the code paths if the TLS secret and cert dir aren't supplied? If I follow correctly:

* No webhooks would actually be registered (so we wouldn't be listening for webhooks?)
* We would still accept webhook configs in provider packages.
* We'd still create a secret, if a revision somehow had a TLS secret name set.

Reply:
I have added automatic TLS secret generation so the only parameter we expect from user is to set `--set webhooks.enabled=true` with Helm installation and its default is `false`. But the code itself depends on whether a secret name is given which is controlled by this toggle. So, from user's perspective it's all the same, it's just we manage the toggle in Helm instead of at the CLI level.

FWIW, the moment Crossplane uses a conversion webhook, users would have to enable this, otherwise initializer rejects proceeding. Similarly, if a package brings in a CRD with conversion webhook strategy and the webhooks weren't enabled in Crossplane, the installation will fail. Both are because those CRDs cannot be used without a certificate injected.